src/backend/distributed/metadata/metadata_utility.c

```diff
@@ -2357,3 +2363,1594 @@ IsForeignTable(Oid relationId)
+		case BACKGROUND_TASK_STATUS_CANCELLED:
+		{
+			return true;
+		}
+
+		case BACKGROUND_TASK_STATUS_RUNNABLE:
+		case BACKGROUND_TASK_STATUS_BLOCKED:
+		case BACKGROUND_TASK_STATUS_RUNNING:
+		{
+			return false;
+		}
+
+			/* no default to make sure we explicitly add every state here */
+	}
+	elog(ERROR, "unknown BackgroundTaskStatus");
+}
+
+
+/*
+ * BackgroundJobStatusOid returns the Oid corresponding to SQL enum value corresponding to
+ * the BackgroundJobStatus.
+ */
+Oid
+BackgroundJobStatusOid(BackgroundJobStatus status)
+{
+	switch (status)
+	{
+		case BACKGROUND_JOB_STATUS_SCHEDULED:
+		{
+			return CitusJobStatusScheduledId();
+		}
+
+		case BACKGROUND_JOB_STATUS_RUNNING:
+		{
+			return CitusJobStatusRunningId();
+		}
+
+		case BACKGROUND_JOB_STATUS_FINISHED:
+		{
+			return CitusJobStatusFinisehdId();
+		}
+
+		case BACKGROUND_JOB_STATUS_CANCELLED:
+		{
+			return CitusJobStatusCancelledId();
+		}
+
+		case BACKGROUND_JOB_STATUS_FAILING:
+		{
+			return CitusJobStatusFailingId();
+		}
+
+		case BACKGROUND_JOB_STATUS_FAILED:
+		{
+			return CitusJobStatusFailedId();
+		}
+	}
+
+	elog(ERROR, "unknown BackgroundJobStatus");
+	return InvalidOid;
+}
+
+
+/*
+ * BackgroundTaskStatusOid returns the Oid corresponding to SQL enum value corresponding to
+ * the BackgroundTaskStatus.
+ */
+Oid
+BackgroundTaskStatusOid(BackgroundTaskStatus status)
+{
+	switch (status)
+	{
+		case BACKGROUND_TASK_STATUS_BLOCKED:
+		{
+			return CitusTaskStatusBlockedId();
+		}
+
+		case BACKGROUND_TASK_STATUS_RUNNABLE:
+		{
+			return CitusTaskStatusRunnableId();
+		}
+
+		case BACKGROUND_TASK_STATUS_RUNNING:
+		{
+			return CitusTaskStatusRunningId();
+		}
+
+		case BACKGROUND_TASK_STATUS_DONE:
+		{
+			return CitusTaskStatusDoneId();
+		}
+
+		case BACKGROUND_TASK_STATUS_ERROR:
+		{
+			return CitusTaskStatusErrorId();
+		}
+
+		case BACKGROUND_TASK_STATUS_UNSCHEDULED:
+		{
+			return CitusTaskStatusUnscheduledId();
+		}
+
+		case BACKGROUND_TASK_STATUS_CANCELLED:
+		{
+			return CitusTaskStatusCancelledId();
+		}
+	}
+
+	elog(ERROR, "unknown BackgroundTaskStatus");
+	return InvalidOid;
+}
+
+
+/*
+ * GetNextBackgroundJobsJobId reads and increments the SQL sequence associated with the
+ * background job's job_id. After incrementing the counter it returns the counter back to
+ * the caller.
+ *
+ * The return value is typically used to insert new jobs into the catalog.
+ */
+int64
+GetNextBackgroundJobsJobId(void)
+{
+	text *sequenceName = cstring_to_text(PG_DIST_BACKGROUND_JOBS_JOB_ID_SEQUENCE_NAME);
+	Oid sequenceId = ResolveRelationId(sequenceName, false);
+	Datum sequenceIdDatum = ObjectIdGetDatum(sequenceId);
+	Oid savedUserId = InvalidOid;
+	int savedSecurityContext = 0;
+
+	GetUserIdAndSecContext(&savedUserId, &savedSecurityContext);
+	SetUserIdAndSecContext(CitusExtensionOwner(), SECURITY_LOCAL_USERID_CHANGE);
+
+	/* generate new and unique colocation id from sequence */
+	Datum jobIdOid = DirectFunctionCall1(nextval_oid, sequenceIdDatum);
```

Comment:
In recent postgres versions, it seems we can call nextval_internal with the second argument false to read from a sequence number without switching to superuser.